lib/AIECreateLogicalLocks.cpp

```diff
@@ -0,0 +1,72 @@
+//===- AIEAssignBuffers.cpp -------------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2019 Xilinx Inc.
+//
+//===----------------------------------------------------------------------===//
+
+// This pass aims to assign lockIDs to AIE.lock operations. The lockID is
+// numbered from the most recent AIE.lock within the same tile. If the lockID
+// exceeds 15 then the pass generates an error and terminates. AIE.lock
+// operations for different tiles are numbered independently. If there exists an
+// existing lockID this pass overwrites the existing lockID generating a
```

Comment:
I think if there are existing lock IDs, this pass should be idempotent and not change anything.  (i.e., it should only assign lock ids to locks without an ID.)